### DISCUSSION TITLE
How the thrust battery compensation is made for bolt base UAV ?

### DISCUSSION THREAD

**Discussion by GusBon** [bitcraze/discussions#1419] [Q&A] [answered]
Hi,

I am currently working with Bolt based UAV with 3S batteries and I am trying to understand better how the attitude control is made.
To do so I am reading firmware and documentation on that part, and currently I am trying to understand the powerDistribution part (from output of the attitude controller to the command of each motor). 
And I have a question about the battery voltage drop made by the motorsCompensateBatteryVoltage() function in motors.c code :
https://github.com/bitcraze/crazyflie-firmware/blob/7cbe7cc56ff52d9b0b7f9442858d801eca35a6bd/src/drivers/src/motors.c#L133-L204

How is this part is made for a UAV that has not a 1S battery ? As the quadratic function is made for 1S battery and some parameters are hardcoded (the critical voltage threshold, and the max thrust in the reduction computation).

Thanks in advance !

**Comment by tobbeanton** [comment, marked as answer]
The motorsCompensateBatteryVoltage() is only enabled for [brushed](https://github.com/bitcraze/crazyflie-firmware/blob/master/src/drivers/src/motors.c#L181) (stock) setup.

If you would like to try and use it for your setup [here](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/functional-areas/pwm-to-thrust/#old-measurements-2015) is how we did it.

**Reply by GusBon** [reply]
Hi !

Thanks for the fast response !

Sorry I didn't saw/understood the if statement line for brushed motors.
I will probably not have the time, the equipment and skills to replicate the process to determine the polynomial function. But this is interesting to know !

Thanks a lot.
